core/src/main/java/io/grpc/transport/MessageFramer.java

```diff
@@ -166,13 +185,41 @@ private void writeFrame(InputStream message, int messageLength, boolean compress
     }
   }
 
+  /**
+   * Write a message that has been serialized to a sequence of buffers.
+   */
+  private void writeBufferChain(BufferChainOutputStream bufferChain, boolean compressed)
+      throws IOException {
+    ByteBuffer header = ByteBuffer.wrap(headerScratch);
```

Comment:
Oh right ... this is our allocator (which can underallocate), not Netty's.  Hmm ... if it were to underallocate, wouldn't writing scratch to it be a problem as well?